### PR TITLE
chore: Define default resource and data source documentation template

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Ticket: https://jira.mongodb.org/browse/CLOUDP-222897

**Context:**
 - Currently all generated documentation requires a new template to be defined.

**Proposed Solution:**
-  Define a generic template for resources as well as single/plural data sources.

For documentation that requires no specific changes the default template can be used and no new template needs to be created. In case specific adjustments are needed, the default template will serve as a starting point to define the new template.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

## Testing
I ran `export resource_name=cluster && make generate-doc` and chcke that the default template was used.

### CODE REVIEW

File: templates/data-source.html.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    {{ else if eq .Name "mongodbatlas_encryption_at_rest" }}
+    {{ else if eq .Name "mongodbatlas_ldap_configuration" }}
+    {{ else if eq .Name "mongodbatlas_teams" }}
+    {{ else if eq .Name "mongodbatlas_serverless_instance" }}
+    {{ else if eq .Name "mongodbatlas_maintenance_window" }}
+    {{ else if eq .Name "mongodbatlas_ldap_verify" }}
+    {{ else if eq .Name "mongodbatlas_third_party_integration" }}
+    {{ else if eq .Name "mongodbatlas_x509_authentication_database_user" }}
+    {{ else if eq .Name "mongodbatlas_privatelink_endpoint_service_data_federation_online_archive" }}
+    {{ else }}
```
Comment (marcosuma) [inline templates/data-source.html.md.tmpl:68]: could you explain why we need all these `else if` and not simply one `else` like the one at the bottom?

Comment (andreaangiolillo) [reply templates/data-source.html.md.tmpl:68]: sure, these resources do not have an example in the `examples` folder and the gen doc tool fails if the example is not there. The last else instead is accessing the example in the "default" location (`examples/[resource_name]/main.tf`)

Comment (marcosuma) [reply templates/data-source.html.md.tmpl:68]: ahhh gotcha ok, I thought all those else if would end up in the last statemnet. ok for me

**Review by marcosuma — APPROVED** [review]

File: templates/resources.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {{ tffile (printf "examples/%s/main.tf" .Name )}}
+    {{ end }}
+{{ end }}
+
+{{ if .SchemaMarkdown }}{{ .SchemaMarkdown | trimspace }}{{ end }}
+
+# Import 
+## Add the import
+
+For more information see: [MongoDB Atlas API - Search Node](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Atlas-Search/operation/createAtlasSearchDeployment) Documentation.
```
Comment (AgustinBettati) [inline templates/resources.md.tmpl:77]: I believe we can remove this from the default template right? Not sure if we are capable of leaving this generic. Applies to the data source template as well.

**Review by AgustinBettati — COMMENTED** [review]
A small detail

File: templates/resources.md.tmpl (hunk 2 above)
Comment (andreaangiolillo) [reply templates/resources.md.tmpl:77]: > I believe we can remove this from the default template right? Not sure if we are capable of leaving this generic.

we can have the import autogenerated if we define an `import.txt` in the example folder. similar to https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1394/files#diff-ab6cba42e3e4a9f3642faec48e42d41f668902c8ffa3041a36bbc56e01c59913. What do you think? 

> applies to the data source template as well.

data sources do not have the import

**Pull request merged**